### PR TITLE
[#5076] previous cumulative update endpoint

### PR DESCRIPTION
**Pull request by zuhdil**
# TODO / Done

Make an endpoint to get the previous cumulative update by the logged-in user

 - [x] Add Indicator.cumulative field
 - [x] Make endpoint

# Test plan

What tests are necessary to ensure this works or doesn't break anything working

 - [x] Unit test

### CODE REVIEW

File: akvo/rsr/tests/usecases/test_previous_cumulative_update_by_user.py [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+                                    'period_start': date(2011, 1, 1),
+                                    'period_end': date(2011, 12, 31),
+                                },
+                            ]
+                        }
+                    ]
+                }
+            ])\
+            .build()
+        self.indicator = self.project.indicators.get(title='Indicator #1')
```
Comment (MichaelAkvo) [inline akvo/rsr/tests/usecases/test_previous_cumulative_update_by_user.py:84]: This seems to be the same in all added classes. A base class for cumulative updates could be used.

**Review by MichaelAkvo — CHANGES_REQUESTED** [review]

File: akvo/rsr/usecases/previous_cumulative_update_by_user.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        'disaggregations': {},
+    }
+
+
+def get_previous_cumulative_update_value(user, indicator):
+    if not indicator.cumulative:
+        return empty_result()
+    latest_update = IndicatorPeriodData.objects\
+        .prefetch_related('disaggregations', 'disaggregations__dimension_value', 'disaggregations__dimension_value__name')\
+        .filter(user=user, period__indicator=indicator, status=IndicatorPeriodData.STATUS_APPROVED_CODE)\
```
[the remark this answers is not in the document]
Comment (zuhdil) [reply akvo/rsr/usecases/previous_cumulative_update_by_user.py:18]: The endpoint will be used in the enumerator form to hint at the enumerator of their previously submitted update. It is less useful to show other user updates and can lead to confusion.

File: akvo/rsr/tests/usecases/test_previous_cumulative_update_by_user.py (hunk 1 above)
Comment (zuhdil) [reply akvo/rsr/tests/usecases/test_previous_cumulative_update_by_user.py]: fixed

File: akvo/rsr/usecases/previous_cumulative_update_by_user.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        .order_by('-period__period_end', '-created_at')\
+        .first()
+    if not latest_update:
+        return empty_result()
+    disaggregations = {}
+    for d in latest_update.disaggregations.all():
+        category = d.dimension_value.name.name
+        type = d.dimension_value.value
+        disaggregations.setdefault(category, {}).setdefault(type, {
```
[the remark this answers is not in the document]
Comment (zuhdil) [reply akvo/rsr/usecases/previous_cumulative_update_by_user.py]: fixed

File: akvo/rsr/models/result/indicator.py [hunk 4]
```diff
@@ -63,6 +63,10 @@ class Indicator(models.Model):
                     'baseline value (eg. people with access to sanitation). Choose descending if '
                     'the target value of the indicator is lower than the baseline value '
                     '(eg. people with diarrhea).'))
+    cumulative = models.BooleanField(
+        _('cumulative'), default=False,
+        help_text=_('Select if the reporting update is cumulative')
```
[the remark this answers is not in the document]
Comment (zuhdil) [reply akvo/rsr/models/result/indicator.py]: fixed

**Review by MichaelAkvo — APPROVED** [review]
:+1: Looks good to me now

**Pull request merged**